### PR TITLE
Ee 18400

### PR DESCRIPTION
**Pull request by Tim-Hammonds**
Preparatory change, prior to the real change for IncomeSummaryService and HmrcClient

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/pttg/api/RequestHeaderData.java [hunk 1]
```diff
@@ -105,33 +115,43 @@ private void initialiseUserName(HttpServletRequest request) {
...
+    }
 
-            try {
-                maxDuration = Integer.parseInt(header);
-            } catch (NumberFormatException e) {
-                throw new IllegalArgumentException(String.format("Cannot parse %s header (%s) into Integer", MAX_DURATION_MS_HEADER, header), e);
-            }
-        }
+    private int maxDuration(String header) {
+        log.info("In maxDuration setter with header {}", header);
```
Comment (owainrobertsbjss) [inline src/main/java/uk/gov/digital/ho/pttg/api/RequestHeaderData.java]: This looks like it should be debug level. It has no Event either

File: src/main/java/uk/gov/digital/ho/pttg/api/RequestHeaderData.java [hunk 2]
```diff
@@ -105,33 +115,43 @@ private void initialiseUserName(HttpServletRequest request) {
...
 
-        log.info("Hmrc service response required in {} ms", maxDuration, value(EVENT, HMRC_SERVICE_MAX_RESPONSE_TIME));
+        if (StringUtils.isBlank(header)) {
+            log.info("In maxDuration setter no header so using default {}", defaultMaxDuration);
```
Comment (owainrobertsbjss) [inline src/main/java/uk/gov/digital/ho/pttg/api/RequestHeaderData.java]: Also looks like a debug log

File: src/main/java/uk/gov/digital/ho/pttg/application/HmrcRetryTemplateFactory.java [hunk 3]
```diff
@@ -45,12 +47,14 @@ private CompositeRetryPolicy compositeRetryPolicy(SimpleRetryPolicy simpleRetryP
     }
 
     private SimpleRetryPolicy simpleRetryPolicy(int attempts) {
+        log.info("Retry policy has {} attempts", attempts);
```
Comment (owainrobertsbjss) [inline src/main/java/uk/gov/digital/ho/pttg/application/HmrcRetryTemplateFactory.java]: Needs an event ID

File: src/main/java/uk/gov/digital/ho/pttg/application/HmrcRetryTemplateFactory.java [hunk 4]
```diff
@@ -45,12 +47,14 @@ private CompositeRetryPolicy compositeRetryPolicy(SimpleRetryPolicy simpleRetryP
...
         return new SimpleRetryPolicy(
                 attempts,
                 singletonMap(HttpServerErrorException.class, TRUE));
     }
 
     private TimeoutRetryPolicy timeoutRetryPolicy(int maxDurationInMs) {
+        log.info("Retry policy has max duration of {} milliseconds", maxDurationInMs);
```
Comment (owainrobertsbjss) [inline src/main/java/uk/gov/digital/ho/pttg/application/HmrcRetryTemplateFactory.java]: Needs an event ID

File: src/test/java/uk/gov/digital/ho/pttg/api/RequestHeaderDataTest.java [hunk 5]
```diff
@@ -63,89 +78,115 @@ public void shouldDefaultRequestData() {
...
+                .willReturn(null);
 
-        requestData.preHandle(mockHttpServletRequest, mockHttpServletResponse, mockHandler);
+        when_requestDataPrehandleCalled();
 
-        verify(mockAppender).doAppend(argThat(argument -> {
-            LoggingEvent loggingEvent = (LoggingEvent) argument;
+        then(mockAppender)
+                .should()
+                .doAppend(argThat(argument -> {
```
Comment (owainrobertsbjss) [inline src/test/java/uk/gov/digital/ho/pttg/api/RequestHeaderDataTest.java]: Changing to the then.should() format has brought the indentation in even further here and it's now quite ugly. Could the lambda perhaps be extracted to a method?

File: src/main/java/uk/gov/digital/ho/pttg/application/HmrcRetryTemplateFactory.java (hunk 3 above)
Comment (Tim-Hammonds) [reply src/main/java/uk/gov/digital/ho/pttg/application/HmrcRetryTemplateFactory.java]: I didn't view this as an "event", hence no event ID. It is really debug, but debug doesn't get picked up in prod ElasticSearch.

**Pull request closed without merging**